$OpenBSD: patch-sh_lex_c,v 1.1 2021/03/27 16:54:23 naddy Exp $

Restore the behavior where both $foo:h and $foo:t return the whole string
https://github.com/tcsh-org/tcsh/commit/05d3d6bfc22462584f9347493499621056995ac1

Index: sh.lex.c
--- sh.lex.c.orig
+++ sh.lex.c
@@ -1046,7 +1046,7 @@ domod(Char *cp, Char type)
     case 't':
 	wp = Strrchr(cp, '/');
 	if (wp == NULL)
-	    return Strsave(type == 't' ? cp : STRNULL);
+	    return NULL;
 	if (type == 't')
 	    xp = Strsave(wp + 1);
 	else
